lib/std/terminfo.zig

```diff
@@ -0,0 +1,1122 @@
+    .{ .short = "colornm", .long = "color_names" },
+    .{ .short = "defbi", .long = "define_bit_image_region" },
+    .{ .short = "devt", .long = "device_type" },
+    .{ .short = "dispc", .long = "display_pc_char" },
+    .{ .short = "endbi", .long = "end_bit_image_region" },
+    .{ .short = "smpch", .long = "enter_pc_charset_mode" },
+    .{ .short = "smsc", .long = "enter_scancode_mode" },
+    .{ .short = "rmpch", .long = "exit_pc_charset_mode" },
+    .{ .short = "rmsc", .long = "exit_scancode_mode" },
+    .{ .short = "getm", .long = "get_mouse" },
+    .{ .short = "kmous", .long = "key_mouse" },
+    .{ .short = "minfo", .long = "mouse_info" },
+    .{ .short = "pctrm", .long = "pc_term_options" },
+    .{ .short = "pfxl", .long = "pkey_plab" },
+    .{ .short = "reqmp", .long = "req_mouse_pos" },
+    .{ .short = "scesc", .long = "scancode_escape" },
+    .{ .short = "s0ds", .long = "set0_des_seq" },
+    .{ .short = "s1ds", .long = "set1_des_seq" },
+    .{ .short = "s2ds", .long = "set2_des_seq" },
+    .{ .short = "s3ds", .long = "set3_des_seq" },
+    .{ .short = "setab", .long = "set_a_background" },
+    .{ .short = "setaf", .long = "set_a_foreground" },
+    .{ .short = "setcolor", .long = "set_color_band" },
+    .{ .short = "smglr", .long = "set_lr_margin" },
+    .{ .short = "slines", .long = "set_page_length" },
+    .{ .short = "smgtb", .long = "set_tb_margin" },
+    .{ .short = "ehhlm", .long = "enter_horizontal_hl_mode" },
+    .{ .short = "elhlm", .long = "enter_left_hl_mode" },
+    .{ .short = "elohlm", .long = "enter_low_hl_mode" },
+    .{ .short = "erhlm", .long = "enter_right_hl_mode" },
+    .{ .short = "ethlm", .long = "enter_top_hl_mode" },
+    .{ .short = "evhlm", .long = "enter_vertical_hl_mode" },
+    .{ .short = "sgr1", .long = "set_a_attributes" },
+    .{ .short = "slength", .long = "set_pglen_inch" },
+};
+
+pub fn getCapsLookUpTable() !std.StringHashMap([]const u8) {
+    var data = std.StringHashMap([]const u8).init(std.heap.page_allocator);
+    for (dictionary) |term| 
+        _ = try data.put(term.short, term.long);
+    return data;
+}
+
+pub fn ReadLineIterator(comptime buffer_size: usize) type {
+    return struct {
+        const Self = @This();
+        const Stream = std.io.InStream(std.os.ReadError);
+        const FifoType = std.fifo.LinearFifo(u8, std.fifo.LinearFifoBufferType{ .Static = buffer_size });
+        const FilterCallback = fn([]u8)bool;
+        const MapCallback = fn([]u8)[]const u8;
+
+        fifo: FifoType,
+        file: std.fs.File,
+        stream: Stream,
+        filter_fn: FilterCallback,
+        map_fn: MapCallback,
+
+        pub fn init(file_path: []const u8) !Self {
+            const file = try std.fs.cwd().openFile(file_path, .{});
+            var in_stream = file.inStream();
+
+            return Self {
+                .fifo = FifoType.init(),
+                .file = file,
+                .stream = in_stream.stream,
```

Comment:
You're taking the stream out of the file instream struct here. You need to keep the whole `in_stream` object around.